am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java

```diff
@@ -241,8 +250,29 @@ public FilteredResourceEnvelope filterResource(@NotBlank String userId,
     @SuppressWarnings("LineLength")
     @AuditLog("returned resources that user with roles '{{userRoles}}' has create permission to: {{result}}")
     public Set<ResourceDefinition> getResourceDefinitionsWithRootCreatePermission(@NotEmpty Set<@NotBlank String> userRoles) {
-        return jdbi.withExtension(AccessManagementRepository.class, dao ->
-            dao.getResourceDefinitionsWithRootCreatePermission(userRoles));
+        Set<ResourceAttribute> resourceAttributes = jdbi.withExtension(AccessManagementRepository.class, dao ->
```

Comment:
I am happy to change it to this way - I guess this way there would be less calls to the db... so it would in theory be quicker. I will run java benchmark on each way and see what works best